### DISCUSSION TITLE
How to prevent the page auto scroll to top when using router.push

### DISCUSSION THREAD

**Discussion by kaisiant** [vercel/next.js#22254] [Help] [answered]
Refer discussion #14534, 

Here is my use case, after clicked the pagination I want to change the url query, so below is the code:
```

 router.push(
      {
        pathname: currentPath,
        query: currentQuery,
      },
      undefined,
      { shallow: true }
    )
```

Problem:
The page is auto scroll to top after clicked the button, it is so annoying.

Is there anyway to fix it?

**Comment by michrome** [comment, marked as answer]
Have you tried `{ scroll: false, shallow: true }`?

I have created [a pull request](https://github.com/vercel/next.js/pull/22275) to document this.
